File: packages/typescript/src/endpoints/scan_tools.ts
/**
 * Scan_Tools API Client
 */

import { AxiosInstance } from 'axios'

export class Scan_ToolsClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * GET /tools
   */
  async getTools(active_only?: boolean): Promise<List[ScanToolInfo]> {
        const params = {'active_only: active_only'}
        const data = undefined
        return this.httpClient.get(`/tools`, params)
  }
  /**
   * GET /tools/{tool_name}
   */
  async getTools{tool_name}(tool_name: string): Promise<ScanToolInfo> {
        const params = {'tool_name: tool_name'}
        const data = undefined
        return this.httpClient.get(`/tools/{tool_name}`, params)
  }
  /**
   * GET /templates
   */
  async getTemplates(tool?: string, category?: string, language?: string, active_only?: boolean): Promise<List[ScanTemplateInfo]> {
        const params = {'tool: tool, category: category, language: language, active_only: active_only'}
        const data = undefined
        return this.httpClient.get(`/templates`, params)
  }
  /**
   * GET /templates/{template_id}
   */
  async getTemplates{template_id}(template_id: string): Promise<ScanTemplateInfo> {
        const params = {'template_id: template_id'}
        const data = undefined
        return this.httpClient.get(`/templates/{template_id}`, params)
  }
  /**
   * POST /validate-configuration
   */
  async postValidateconfiguration(): Promise<ScanConfigurationValidationResponse> {
        const params = {}
        const data = undefined
        return this.httpClient.post(`/validate-configuration`, data, params)
  }
  /**
   * GET /repositories/{repository_id}/settings
   */
  async getRepositories{repository_id}settings(repository_id: string): Promise<RepositoryScanSettings> {
        const params = {'repository_id: repository_id'}
        const data = undefined
        return this.httpClient.get(`/repositories/{repository_id}/settings`, params)
  }
  /**
   * PUT /repositories/{repository_id}/settings
   */
  async putRepositories{repository_id}settings(repository_id: string, settings: RepositoryScanSettings): Promise<RepositoryScanSettings> {
        const params = {}
        const data = {'repository_id: repository_id, settings: settings'}
        return this.httpClient.put(`/repositories/{repository_id}/settings`, data, params)
  }
  /**
   * POST /validate-access
   */
  async postValidateaccess(): Promise<RepositoryAccessValidationResponse> {
        const params = {}
        const data = undefined
        return this.httpClient.post(`/validate-access`, data, params)
  }
}

File: packages/python/src/endpoints/scan_tools.py
"""
Scan_Tools API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class Scan_ToolsClient:
    """Client for scan_tools endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def get_tools(self, active_only: Optional[bool] = None):
        """GET /tools"""
        url = f"/tools"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_tools_tool_name(self, tool_name: str):
        """GET /tools/{tool_name}"""
        url = f"/tools/{tool_name}"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_templates(self, tool: Optional[str] = None, category: Optional[str] = None, language: Optional[str] = None, active_only: Optional[bool] = None):
        """GET /templates"""
        url = f"/templates"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_templates_template_id(self, template_id: str):
        """GET /templates/{template_id}"""
        url = f"/templates/{template_id}"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_validate_configuration(self):
        """POST /validate-configuration"""
        url = f"/validate-configuration"
        params = {}
        data = None
        return await self._request("POST", url, params=params, json=data)


    async def get_repositories_repository_id_settings(self, repository_id: str):
        """GET /repositories/{repository_id}/settings"""
        url = f"/repositories/{repository_id}/settings"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def put_repositories_repository_id_settings(self, repository_id: str, settings: Any):
        """PUT /repositories/{repository_id}/settings"""
        url = f"/repositories/{repository_id}/settings"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("PUT", url, params=params, json=data)


    async def post_validate_access(self):
        """POST /validate-access"""
        url = f"/validate-access"
        params = {}
        data = None
        return await self._request("POST", url, params=params, json=data)

